Random forest reprogramming blog post (#14)
czbiohub/blog: _blog/random-forest-reprogramming.md

---
layout: post
mathjax: true
codehide: true
title: How can scRNA-seq inform the development of new reprogramming protocols and sorting markers pannels?
date: 2018-10-26
author: Angela Oliveira Pisco
---

## From transcription factors to reprogramming protocols

> [Induced pluripotent stem cells](https://www.eurostemcell.org/ips-cells-and-reprogramming-turn-any-cell-body-stem-cell) are the holy grail of regenerative medicine; however, the low efficiency of somatic cell reprogramming and the lengthy cell differentiation protocols are hardening their potential clinical applications. For this reason, one of the current major goals of single cell research is defining cell identities because there lies the key to the development of better reprogramming protocols. Critical to such work is the understanding of the underlying regulatory networks. While a handful of molecular functions are usually associated with cell regulation, the most commonly used are transcription factors or transcription regulators in general. In [Tabula muris](https://www.nature.com/articles/s41586-018-0590-4) we investigated how transcription factors are able to reconstruct known cell ontology relationships between cell populations. Using a random forest model strategy we identified sets of transcription factors using [facs](https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-018-0590-4/MediaObjects/41586_2018_590_MOESM8_ESM.xlsx) and [droplet](https://github.com/czbiohub/tabula-muris/blob/master/23_tf_analysis/rf.model.one.vs.all.transcriptionfactors.droplet.xlsx) data that can potentially be used to inform the design of novel reprogramming protocols.

Reviewer: The first half feels very paper-y to me–the post itself is short and sweet. I don't think you need to sell the reader much on the broader impact. Maybe as a conclusion?

Reviewer: Also, this is the teaser paragraph on the main page, so keeping it to a concise summary of the post would be nice

> ![Random forest model using transcription factors](/images/reprogramming-direct-diff/rf_tfs_summary.png)

Reviewer: A caption would be helpful to explain this, or just some text to introduce us to the figure before we see it.

A couple typos: "i~n~teratively",  "transcription~s~ factors"

## Contribution of transcription factors to cell identity

> Our confidence in the model comes from empirical validation, when comparing the top 10 candidates with the transcription factors currently used to reprogram some of the cell types. The choice of using transcription factors to pull each cell type a part from the rest was natural given that using only transcription factors we were able to reconstruct the full dendrogram of cell identities with 90\% confidence

Reviewer: ```suggestion
Our confidence in the model comes from comparing the top 10 candidates with the transcription factors currently used to reprogram some of the cell types. The choice to use transcription factors to pull each cell type apart from the rest was natural, given that using only transcription factors we were able to reconstruct the full dendrogram of cell identities with 90% confidence.
```

Still reads a little weird to me, not sure how to fix

> ![Tanglegram transcription factors](/images/reprogramming-direct-diff/rf_tfs_entanglements.png)

Reviewer: Again, having a caption so I had any idea what this was showing me would be useful...

Reviewer: Despite being huge it's also kind of unreadable. And it's a supplement to the paper itself (right?) so maybe it can just be linked to.

> in contrast with many other chosen functional groups.

Reviewer: this sentence fragment is all alone...I didn't realize it was part of the previous section! There's a huge figure between them

> ![Entanglements](/images/reprogramming-direct-diff/rf_entanglements.png)

Reviewer: Caption please! What is "entanglement"? Also the text on the bars is pretty hard to read

## Cell surface markers

> While it is of note the inability of cell surface markers to pull cell types apart, the cell surface antigens are among the most ubiquitous class of proteins used on a daily basis in any cell biology lab. The majority of cell surface markers are molecules within cell's plasma membrane that are unique to different cell types. We sought to try using the random forest model used for identifying transcription factors associated with individual cell identities to come up with cell sorting panels using the [facs](https://github.com/czbiohub/tabula-muris/blob/master/23_tf_analysis/rf.model.one.vs.all.cellsurfacemarkers.facs.xlsx) and the [droplet](https://github.com/czbiohub/tabula-muris/blob/master/23_tf_analysis/rf.model.one.vs.all.cellsurfacemarkers.droplet.xlsx) data in Tabula muris.

Reviewer: ```suggestion
While the inability of cell surface markers to pull cell types apart is notable, the cell surface antigens are some of the most commonly used proteins in any cell biology lab. The majority of cell surface markers are molecules within the cell's plasma membrane that are unique to different cell types. We tried the same random forest model to come up with cell sorting panels using the [FACS](https://raw.githubusercontent.com/czbiohub/tabula-muris/blob/master/23_tf_analysis/rf.model.one.vs.all.cellsurfacemarkers.facs.xlsx) and the [droplet](https://raw.githubusercontent.com/czbiohub/tabula-muris/blob/master/23_tf_analysis/rf.model.one.vs.all.cellsurfacemarkers.droplet.xlsx) data in the Tabula Muris.
```

I don't think I understand this paragraph, though–they're unable to pull types apart, but we used them to come up with markers anyway?

> ## Seeking collaborations

Reviewer: I don't think we want to say this here

Reviewer: These are interesting future directions but we're not putting out a call for collaborators, ain't nobody got that kind of time

[missing conclusion here]

## Conclusion
[Induced pluripotent stem cells](https://www.eurostemcell.org/ips-cells-and-reprogramming-turn-any-cell-body-stem-cell) are the holy grail of regenerative medicine; however, the low efficiency of somatic cell reprogramming and the lengthy cell differentiation protocols are hardening their potential clinical applications. For this reason, one of the current major goals of single cell research is defining cell identities because there lies the key to the development of better reprogramming protocols. With our machine learning approach we are not only able to provide candidate transcription factors for novel reprogramming protocols, but we can already take a step further and suggest cell type validation sets, as our model can also predict cell surface markers that be used either for FACS sorting or antibody staining.

> While this is still very experimental, the results match previously known findings, as for example the combination of *Cd36* and *Cav1* being enough to sort fat endothelial cells. We are interested in knowing whether our model predictions are valid in cell types not previously reported, so if would like to take this further and experimental validate the results by either testing the transcription factors or the cell surface markers candidates reach out!

Reviewer: ```suggestion
While this is still very experimental, the results match previously known findings, for example the combination of *Cd36* and *Cav1* being enough to sort fat endothelial cells. We are interested in knowing whether our model predictions are valid in cell types not previously reported, so if you would like to take this further and experimentally test these predictions, reach out!
```

I'm still not sure we want to make that invitation but it's your project I guess...
